python/cudf/udf_cpp/groupby/function.cu

```diff
@@ -28,20 +28,22 @@ __device__ bool are_all_nans(cooperative_groups::thread_block const& block,
                              T const* data,
                              int64_t size)
 {
-  __shared__ bool result;
+  // TODO: to be refactored with CG vote functions once
+  // block size is known at build time
+  __shared__ int result;
 
-  if (block.thread_rank() == 0) { result = true; }
+  if (block.thread_rank() == 0) { result = 0; }
   block.sync();
 
   for (int64_t idx = block.thread_rank(); idx < size; idx += block.size()) {
     if (not std::isnan(data[idx])) {
-      result = false;
+      atomicAdd(&result, 1);
```

Comment:
How big can `size` be (it's passed as an `int64_t`)? Because if it can be larger than `std::numeric_limits<int>::max` then this is more UB.

Reply:
Use `atomic_ref<int64_t>` instead

Reply:
Thanks!